mkFit/PropagationMPlex.cc

```diff
@@ -666,28 +667,21 @@ void helixAtZ(const MPlexLV& inPar,  const MPlexQI& inChg, const MPlexQF &msZ,
       dprint_np(n, std::endl << "outPar.At(n, 0, 0)=" << outPar.At(n, 0, 0) << " outPar.At(n, 1, 0)=" << outPar.At(n, 1, 0)
 		<< " pxin=" << pxin << " pyin=" << pyin);
 
-      const float sCosPsinah = std::sin(cosP*sina*0.5f);
-      const float cCosPsinah = std::cos(cosP*sina*0.5f);
-      const float sCosPsina = 2.f*sCosPsinah*cCosPsinah;
-      const float sPsCPless1 = sinP*sCosPsina - 1.f;
-      const float cAlessFsAA = alpha != 0 ? cosa - sina/alpha : 0.f; // lim-> alpha^2/3
-      const float sCPx2 = alpha != 0 ? 2.f*sCosPsinah*(cosP*cosa*cCosPsinah - sCosPsinah/alpha) : 0.f;// lim-> ~alpha
-
-      errorProp(n,0,2) = cosP*tanT*cosa*sPsCPless1;
-      //      errorProp(n,0,3) = cosP*tanT*deltaZ*cosa*( 1.f - sinP*sCosPsina )*pt - k*(cosP*sina - sinP*(1.f-cCosPsina))*pt*pt;
-      errorProp(n,0,3) = tanT*deltaZ*pt*( cosP*cAlessFsAA - sinP*sCPx2 );
-      errorProp(n,0,4) = (k*pt)*( sinP*sina*sPsCPless1 - 2.f*cosP*sCosPsinah*sCosPsinah );
-      errorProp(n,0,5) = -cosP*deltaZ*cosa*sPsCPless1/(cosT*cosT);
-
-      errorProp(n,1,2) = cosa*tanT*(cosP*cosP*sCosPsina - sinP);
-      //      errorProp(n,1,3) = tanT*deltaZ*cosa*( cosP*cosP*sCosPsina + sinP )*pt - k*(sinP*sina + cosP*(1.f-cCosPsina))*pt*pt;
-      errorProp(n,1,3) = tanT*deltaZ*pt*( sinP*cAlessFsAA + cosP*sCPx2 );
-      errorProp(n,1,4) = (k*pt)*( -cosP*sina*sPsCPless1 - 2.f*sinP*sCosPsinah*sCosPsinah );
-      errorProp(n,1,5) = deltaZ*cosa*( cosP*cosP*sCosPsina + sinP )/(cosT*cosT);
+      const float pxcaMpysa = pxin*cosa - pyin*sina;
+      errorProp(n,0,2) = -tanT * ipt * pxcaMpysa;;
+      errorProp(n,0,3) = 2*k*pt*pt*sinah*(sinP*sinah - cosP*cosah) + deltaZ*tanT*pxcaMpysa;
```

Comment:
based on our discussion on skype, this and (1,3) term may need to be refactored to group the large terms (using `deltaZ*tanTheta = k*pt*alpha`)
here `- 2*k*pt*pt*sinah*cosP*cosah = - k*pt*pt*cosP*sina` in the first large term is supposed to [almost] cancel out with the right hand side `deltaZ*tanT*pxin*cosa = k*pt*pt*alpha*cosP*cosa` because `- sina + alpha*cosa` goes to zero; so, it would be best to have this subtraction explicit instead of shuffled with other terms to avoid  loss in numerical precision.

Reply:
... that's for the follow up PR